src/storage/core.rs

```diff
@@ -415,100 +417,88 @@ where
+            let entries = read.unwrap();
             debug!(
                 "storage core read all active blob entries {}",
                 entries.len()
             );
             all_entries.extend(entries);
+        } else if read.is_deleted() {
+            return Ok(read);
         }
         let blobs = safe.blobs.read().await;
-        let entries_closed_blobs = blobs
-            .iter_possible_childs(key)
+        let mut futures = blobs
+            .iter_possible_childs_rev(key)
             .map(|b| b.1.data.read_all_entries(key))
-            .collect::<FuturesUnordered<_>>();
-        entries_closed_blobs
-            .try_filter_map(future::ok)
-            .try_for_each(|v| {
-                debug!("storage core read all closed blob {} entries", v.len());
-                all_entries.extend(v);
-                future::ok(())
-            })
-            .await?;
-        debug!("storage core read all total {} entries", all_entries.len());
-        Ok(all_entries)
+            .collect::<FuturesOrdered<_>>();
+        while let Some(data) = futures.next().await {
+            let data = data?;
+            if data.is_found() {
+                all_entries.extend(data.unwrap());
+            } else if data.is_deleted() {
+                if all_entries.is_empty() {
+                    return Ok(data);
+                } else {
+                    return Ok(ReadResult::Found(all_entries));
+                }
+            }
+        }
+        debug!(
+            "storage core read from non-active total {} entries",
+            all_entries.len()
+        );
+        debug_assert!(all_entries
+            .iter()
+            .zip(all_entries.iter().skip(1))
+            .all(|(x, y)| x.header().created() >= y.header().created()));
+        Ok(ReadResult::Found(all_entries))
     }
 
-    async fn read_with_optional_meta(&self, key: &K, meta: Option<&Meta>) -> Result<Bytes> {
+    async fn read_with_optional_meta(
+        &self,
+        key: &K,
+        meta: Option<&Meta>,
+    ) -> Result<ReadResult<Bytes>> {
         debug!("storage read with optional meta {:?}, {:?}", key, meta);
         let safe = self.inner.safe.read().await;
         if let Some(ablob) = safe.active_blob.as_ref() {
             match ablob.read().await.read_any(key, meta, true).await {
                 Ok(data) => {
-                    debug!("storage read with optional meta active blob returned data");
-                    return Ok(data);
+                    if data.is_presented() {
+                        debug!("storage read with optional meta active blob returned data");
+                        return Ok(data);
+                    }
                 }
                 Err(e) => debug!("read with optional meta active blob returned: {:#?}", e),
             }
         }
         Self::get_any_data(&safe, key, meta).await
     }
 
-    async fn get_data_last(safe: &Safe<K>, key: &K, meta: Option<&Meta>) -> Result<Bytes> {
```

Comment:
Why it was removed? This is the important function that should be used to read data